### ISSUE TITLE
How is load balance loss calculated when gates gradients are detached?

### ISSUE THREAD

**Issue by Rapisurazurite** [yushuiwx/Mixture-of-LoRA-Experts#8]
I noticed in `lora.py` that the gate values are detached when updating `gates_record`:

```python
self.gates_record += gates.clone().detach().mean(dim=0)
```

However, `gates_record` is defined as an `nn.Parameter`:

```python
self.gates_record = nn.Parameter(torch.zeros([num_experts]))
```

My questions are:

1. How is the load balance loss supposed to be calculated in this implementation?
2. Since `gates` gradients are detached, how can the load balance loss backpropagate to update the gating mechanism?
3. Is `gates_record` intended to be used for monitoring purposes only, or should it participate in the loss calculation?

It seems contradictory to use `nn.Parameter` for `gates_record` while detaching the gradients. Could you clarify the intended design here?

Thank you for your clarification!

<img width="844" alt="Image" src="https://github.com/user-attachments/assets/e6e83148-0018-49f8-a13c-becfb3b7e612" />

<img width="903" alt="Image" src="https://github.com/user-attachments/assets/1c1e82c0-7396-4c54-abe6-aec42dc77a68" />

<img width="774" alt="Image" src="https://github.com/user-attachments/assets/efbff81b-2f3b-48e0-ace9-17e00d9da2b6" />
